subprojects/build.gradle.kts

```diff
@@ -270,6 +270,20 @@ subprojects {
 
     tasks.withType<Test> {
         systemProperty("rootDir", "${project.rootDir}")
+
+        val testProperties = listOf(
+            "avito.slack.test.channel",
+            "avito.slack.test.token",
+            "avito.slack.test.workspace"
+        )
+        testProperties.forEach { key ->
+            val property = if (project.hasProperty(key)) {
+                project.property(key)!!.toString()
+            } else {
+                ""
```

Comment:
else `throw IllegalStateException("setup a property")

Reply:
can't do it like that;

It is only applicable for integration (not unit) tests that relies on real slack token;
and i have no idea how to narrow the scope of configuration without duplication

see https://github.com/avito-tech/avito-android/pull/621#discussion_r519605469